packages/domain-search/src/ui/domain-search-controls/filter-popover.tsx

```diff
@@ -18,20 +18,20 @@ import './filter-popover.scss';
 
 type Props = {
 	availableTlds: string[];
+	filter: FilterState;
+	onApply: ( filter: FilterState ) => void;
 	onClear: () => void;
-	onClose: () => void;
-	setTemporaryFilter: ( filter: FilterState ) => void;
-	temporaryFilter: FilterState;
 };
 
 export const DomainSearchControlsFilterPopover = ( {
 	availableTlds,
+	filter,
+	onApply,
 	onClear,
-	onClose,
-	setTemporaryFilter,
-	temporaryFilter,
 }: Props ) => {
 	const { __ } = useI18n();
+	// This is the filter that the user is currently selecting. It is only applied when the popover is closed
+	const [ temporaryFilter, setTemporaryFilter ] = useState( filter );
```

Comment:
Will this change once `filter` changes?

Reply:
I believe so, it worked in my tests 🙂 since the popover component is rebuilt every time it opens, it always uses the most current value of `filter`